src/service/url-expander/expander.js

```diff
@@ -85,6 +85,41 @@ export class Expander {
     return this.parseUrlRecursively_(url, matches);
   }
 
+  /**
+   * Takes an array of binding keys and attempts to substitute values.
+   * The key is an all or nothing subsitution (cannot match only part of a key)
+   * Returns a Promise resolving to an object with keys mapped to substituted
+   * values.  If a key in the input array is not matched, it will not be
+   * included in the output.
+   * @param {Array<string>} keys
+   * @return {Promise<Object<string, string|number>>}
+   */
+  expandBindings(keys) {
```

Comment:
Runtime folk might be not ok with adding extra bytes here to the v0.js.

Reply:
Thanks Dima! 

@jridgewell, not saying this will be needed in the final version, but curious to hear your thoughts on what we should consider, whether or not we should add this?

Reply:
I agree with Dima. This is a lot of code to support `TITLE` and `CANONICAL_URL`. I'd feel much better if we just retrieved them manually.